README.md

```diff
@@ -10,13 +10,62 @@
   <image alt="License" src="https://img.shields.io/github/license/castle/castle-cloudflare-worker-demo"/>
 </div>
 
+## Overview
+
+The Castle Cloudflare Worker allows you to put Castle's risk engine right on the edge, in a Cloudflare worker.
+
+## How it works
+Once you've installed the worker with the default settings, the worker will listen for POSTs to the 
+
+`/users/sign_up`
+
+route.
+
+The POST must include a Castle client_id. Click here to learn more about how to include a Castle client_id in a POST.
+
+When the worker receives the POST, it will in turn make a POST to Castle, and receive a risk score in return. If the risk score is above the `riskThreshold`, then the worker will respond with a 403. If the risk score is below the `riskThreshold`, the worker will respond with a 200.
+
+(In production, you would simply forward the request to the upstream service rather than respond with a 200.)
+
+A simple way to test the service is to change the `riskThreshold` to 0, and then make a request from a different browser.
+
+## Prerequisites
+
+You'll need a Castle account and a Cloudflare account to get started.
+
+### Castle
+
+If you don't have a Castle account already, you can [set up a free trial](https://dashboard.castle.io/signup/new). You will need your Castle API Secret, which can be found in the Settings section of your Castle dashboard.
+
+### Cloudflare
+
+If you're going to use the `Deploy with Workers` option (see below), you'll need your Cloudflare account ID and an API Token. (This integration has been tested with a Cloudflare API Token created with the Cloudflare "Edit Worker" token template.)
+
 ## Installation
 
+There are two options for installing this worker, a "manual" method and a `Deploy with Workers` method.
+
+### Manual installation
+
+1. Create or open the Cloudflare worker where you would like to install the Castle worker code.
+
+2. Add an Environment Variable to your worker:
+
+`CASTLE_API_SECRET` 
+
+and assign your Castle API Key to this variable.
```

Comment:
I think we should provide a link to the Castle dashboard here (as in the section below), as the users might not be aware of where to take the `CASTLE_API_SECRET` from

Reply:
OK.